tests/terraform_pytest/test_terraform_install.py

```diff
@@ -0,0 +1,69 @@
+"""Run Terraform plan/apply against each set of Terraform test files."""
+
+import os
+from pathlib import Path
+
+import pytest
+import tftest
+
+AWS_DEFAULT_REGION = os.getenv("AWS_REGION", default="us-east-1")
+MOCKSTACK_HOST = os.getenv("MOCKSTACK_HOST", default="localhost")
+MOCKSTACK_PORT = "4566"
+
+MOCKSTACK_TF_FILENAME = "mockstack.tf"
+AWS_TF_FILENAME = "aws.tf"
+
+
+@pytest.fixture(scope="function")
+def plan_and_apply(is_mock, tf_dir):
+    """Return the function that will invoke Terraform plan and apply."""
+
+    def invoke_plan_and_apply(tf_module):
+        """Execute Terraform plan and apply."""
+        tf_test = tftest.TerraformTest(tf_module, basedir=str(tf_dir), env=None)
+
+        # Use the appropriate endpoints, either for a simulated AWS stack
+        # or the real deal.
+        provider_tf = MOCKSTACK_TF_FILENAME if is_mock else AWS_TF_FILENAME
+
+        current_dir = Path(__file__).resolve().parent
+        tf_test.setup(extra_files=[str(Path(current_dir / provider_tf))])
+
+        # Update the hostname in the *.tf file to differentiate between using
+        # localhost or using the docker network name.
+        tf_vars = None
+        if is_mock:
+            tf_vars = {
+                "mockstack_host": MOCKSTACK_HOST,
+                "mockstack_port": MOCKSTACK_PORT,
+            }
+
+        # Debugging info:  tftest's plan() will raise an exception if the
+        # return code is 1.  Otherwise, it returns the text of the plan
+        # output.  Adding an argument of "output=True" will return an object
+        # that has members for outputs, resources, modules and variables.
+        # For this test, we're simply looking for pass or fail.
+        #
+        # tftest's apply() will also raise an exception if the return code
+        # is 1.  Otherwise, it returns the output as plain text.
+        try:
+            tf_test.apply(tf_vars=tf_vars)
```

Comment:
Yeah, the test I suggested does confirm the issue with the prereq config being destroyed before the test config runs. Adding `yield` here fixes it.

```suggestion
            yield tf_test.apply(tf_vars=tf_vars)
```

Reply:
This does not work.  I will come up with a solution.  

I can see where the destruction of the prereq plan caused some tests to fail.  If you had free time, you could look at the failing unit tests unrelated to this problem as you most likely wrote those unit tests, but please let me do my job and fix this issue.

Reply:
Alright. Reviews are often about proposing a potential solution to an issue identified in the review. It is just a suggestion to help highlight the problem. Please don't take it any other way.

Reply:
code modified